### PR TITLE
Correct heartbeat message client and server to meet 2022 requirements

### PR DESCRIPTION
**Pull request by alexperez33**
More corrections are needed to simulation environment before we can fully see data from simulation within heartbeat message. However, for now, I corrected the heartbeat client code, along with the code for receiving and sending to td station the entrance/exit gate messages, follow the path messages, wildlife encounter messages, scan the code messages, detect and dock messages, find and fling messages, uav replenishment messages, and the uav search and report messages.

I wrote unittests for each one of these. You can run it with ```rostest navigator_robotx_comms robotx_comms.test```.

The heartbeat message can now be seen from within the simulation environment. The idea is that the simulation environment will be run in the future with ```roslaunch navigator_launch master.launch``` (which contains an arg for simulation or not). If you launch this and then launch ```rosrun navigator_robotx_comms server.py```, you will see any messages that we would send to the td station in competition. All the information is not quite there because there are a few issues that need to be handled outside of this heartbeat issue.

Basically, all these messages we have to send to the td station, I have it set up such that we will have call a ros service and the robotx_comms node will handle gathering, parsing, and sending the information to the td station. We just need to set up the rest of the code to send the proper information to those ros services.

### CODE REVIEW

**Comment by uf-mil-investigator** [comment]
Hello, it's your friendly InvestiGator bot here!

The **docs preview** for this PR is available at https://uf-mil.github.io/pr-docs/831.

```css
Last updated at:
► e82675e99faf4e2ac0490b71ae1e4c1d473354a2
► 2022-08-19 17:05 EDT
```

Have a great day! Go gators! 🐊

**Review by kawaiiPlat — APPROVED** [review]
works on my machine. kool.
